acme.go

```diff
@@ -269,6 +272,77 @@ func pollAuthz(client *http.Client, url string) (*Authorization, error) {
 	return &auth, nil
 }
 
+// newCert requests a new certificate.
+//
+// If client argument is nil, DefaultClient will be used.
+// notBefore and notAfter are optional (if set to zero value)
+func newCert(client *http.Client, config *Config, certKey *rsa.PrivateKey, csr *x509.CertificateRequest, notBefore time.Time, notAfter time.Time) (cert *x509.Certificate, certURL string, err error) {
```

Comment:
Too many arguments :(
Why wouldn't we accept [x509.Certificate](https://golang.org/pkg/crypto/x509/#Certificate) as a template? It has all the fields we need, I think, including `notBefore` and `notAfter`.

Alternatively, accept `csr []byte` and let the caller create it with `x509.CreateCertificateRequest`. I think I prefer the latter because we won't even need to pass private key in.

Either way, `notBefore time.Time, notAfter time.Time` can be shorter: `notBefore, notAfter time.Time`.